services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -290,6 +292,59 @@ def raise_if_too_big_from_datasets(
         )
 
 
+def raise_if_requires_manual_download(
+    dataset: str,
+    config: str,
+    hf_endpoint: str,
+    hf_token: Optional[str],
+    revision: str,
+):
+    """
+    Raise an error if the dataset requires manual download.
+
+    Args:
+        dataset (`str`):
+            A namespace (user or an organization) and a repo name separated
+            by a `/`.
+        config (`str`):
+            Dataset configuration name.
+        hf_endpoint (`str`):
+            The Hub endpoint (for example: "https://huggingface.co").
+        hf_token (`str`, *optional*):
+            An app authentication token with read access to all the datasets.
+        revision (`str`):
+            The git revision (e.g. "main" or sha) of the dataset.
+
+    Returns:
+        `None`
+
+    Raises:
+        [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError):
+            If the datasets.config.HF_ENDPOINT is not set to the expected value.
+        [`libcommon.exceptions.DatasetManualDownloadError`]:
+            If the dataset requires manual download.
+    """
+    if datasets.config.HF_ENDPOINT != hf_endpoint:
```

Comment:
OK, but why here, and not in the rest of the code?

Reply:
I just copied/pasted it from `raise_if_too_big_from_datasets`.

Reply:
ah ah ok, I didn't check!

I think it's weird, and that we should just assume they are equal. Else, we should check every time we use the datasets library. But, OK, it's not for this PR.

Reply:
Yes, I agree.